src/v/iceberg/avro_utils.h

```diff
@@ -45,17 +48,17 @@ class avro_iobuf_ostream : public avro::OutputStream {
         *data = reinterpret_cast<uint8_t*>(
           back_frag->share(chunk_size_ - available_, available_).get_write());
         *len = available_;
-        byte_count_ += available_;
+        *byte_count_ += available_;
         available_ = 0;
         return true;
     }
 
     void backup(size_t len) final {
         available_ += len;
-        byte_count_ -= len;
+        *byte_count_ -= len;
     }
 
-    uint64_t byteCount() const final { return byte_count_; }
+    uint64_t byteCount() const final { return *byte_count_; }
```

Comment:
Can you explain why we need the byte_count to be shared between the caller and this class? It seems like this method is enough, since the caller can always get the byte count.

Reply:
>Can you explain why we need the byte_count to be shared between the caller and this class? It seems like this method is enough, since the caller can always get the byte count.

yeh if the caller needs to customize the intiial value, it oculd pass it in. if it wants to get the final result, we could add an accessor?

Reply:
Sure, the primary user of this code is the DataFileWriter, which uses byteCount() to track its own progress. But it doesn't have any special handling for trimming the stream after writing, there's no close() interface in the stream for the writer to use to indicates it needs to trim. For us, since we have a custom buffer underlying the stream, we need to do that ourselves, which is what this is used for.

Reply:
>there's no close() interface in the stream for the writer to use to indicates it needs to trim. For us, since we have a custom buffer underlying the stream, we need to do that ourselves, which is what this is used for.

makes sense. seems like we could add a wrapper around it that has something like close which we call. but this seems like a reasonable approach given the limitations of the avro interface.

Reply:
Yeah, an accessor would make sense, except that we end up `std::move()`ing the ostream into the DataFileWriter, making use after serializing pretty challenging